hoomd/hpmc/ExternalFieldWall.h

```diff
@@ -481,6 +481,86 @@ inline bool test_confined(const CylinderWall& wall,
     return accept;
     }
 
+// Cylindrical Walls and Convex Spheropolyhedra
+DEVICE inline bool test_confined(const CylinderWall& wall,
+                                 const ShapeSpheropolyhedron& shape,
+                                 const vec3<Scalar>& position,
+                                 const vec3<Scalar>& box_origin,
+                                 const BoxDim& box)
+    {
+    bool accept = true;
+    Scalar3 t = vec_to_scalar3(position - box_origin);
+    t.x = t.x - wall.origin.x;
+    t.y = t.y - wall.origin.y;
+    t.z = t.z - wall.origin.z;
+    vec3<Scalar> shifted_pos(box.minImage(t));
+
+    vec3<Scalar> dist_vec
+        = cross(shifted_pos,
+                wall.orientation); // find the component of the shifted position that is
+                                   // perpendicular to the normalized orientation vector
+    Scalar max_dist = sqrt(dot(dist_vec, dist_vec));
+    if (wall.inside)
```

Comment:
A comment would be helpful to understand what the code is doing here.
```suggestion
    // Check if the shape's bounding sphere intersects the wall.
    if (wall.inside)
```

Also, consider rewriting this so that it evaluates the `<=` or `>` inside the if. The branch is repeated again with a ternary operator on the next line which is very confusing. Better yet, combine the content of the `if check_verts -> if shape.verts.N -> ` branches here as well. Bringing all the "inside" and "outside" code together will make it much easier to understand.